sdk/js-query/src/query/ethCall.test.ts

```diff
@@ -0,0 +1,287 @@
+import {
+  afterAll,
+  beforeAll,
+  describe,
+  expect,
+  jest,
+  test,
+} from "@jest/globals";
+import Web3, { ETH_DATA_FORMAT } from "web3";
+import axios from "axios";
+import * as elliptic from "elliptic";
+import {
+  EthCallData,
+  EthCallQueryRequest,
+  PerChainQueryRequest,
+  QueryRequest,
+  QueryResponse,
+  sign,
+} from "..";
+
+jest.setTimeout(60000);
+
+const CI = false;
+const ENV = "DEVNET";
+const ETH_NODE_URL = CI ? "ws://eth-devnet:8545" : "ws://localhost:8545";
+
+const CCQ_SERVER_URL = "http://localhost:6069/v1";
+const QUERY_URL = CCQ_SERVER_URL + "/query";
+const HEALTH_URL = CCQ_SERVER_URL + "/health";
+const PRIVATE_KEY =
+  "cfb12303a19cde580bb4dd771639b0d26bc68353645571a8cff516ab2ee113a0";
+const WETH_ADDRESS = "0xDDb64fE46a91D46ee29420539FC25FD07c5FEa3E";
+
+let web3: Web3;
+
+beforeAll(() => {
+  web3 = new Web3(ETH_NODE_URL);
+});
+
+afterAll(() => {
+  web3.provider?.disconnect();
+});
+
+function createTestEthCallData(
+  to: string,
+  name: string,
+  outputType: string
+): EthCallData {
+  return {
+    to,
+    data: web3.eth.abi.encodeFunctionCall(
+      {
+        constant: true,
+        inputs: [],
+        name,
+        outputs: [{ name, type: outputType }],
+        payable: false,
+        stateMutability: "view",
+        type: "function",
+      },
+      []
+    ),
+  };
+}
+
+describe("eth call", () => {
+  test("serialize request", () => {
+    const toAddress = "0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270";
+    const nameCallData = createTestEthCallData(toAddress, "name", "string");
+    const totalSupplyCallData = createTestEthCallData(
+      toAddress,
+      "totalSupply",
+      "uint256"
+    );
+    const ethCall = new EthCallQueryRequest("0x28d9630", [
```

Comment:
this is the method selector?

Reply:
jk this looks like a block num?

Reply:
I think making the parameter more explicit like: 
```
// Can be a block number or a block hash
export type BlockTag = number | HexString;
```
would be better